msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -1501,13 +1500,23 @@ public String processActionSaveForumAsDraft()
       setErrorMessage(getResourceBundleString(VALID_FORUM_TITLE_WARN));
       return FORUM_SETTING_REVISE;
     }    
-    saveForumSettings(true);
-    
-    //reset();
-    //return MAIN;
+    if(processForumSettings(true) == null){
+        return null;
+    }
     return processReturnToOriginatingPage();
   }
 
+  private DiscussionForum processForumSettings(boolean draft){
+    if (selectedForum.getForum().getRestrictPermissionsForGroups() && selectedForum.getForum().getId() == null) {
+        if (!saveForumsForGroups(draft)) {
```

Comment:
It would be nice if this method was called from saveForumSettings().

Reply:
It's actually the other way around. As we discussed some time ago, there's a pre-processing needed and then the final saveForumSettings might be called, one or more times. One of the paths actually leads to creating several Forum objects, that's why the saveForumSettings is the inner method.